terminology.md

```diff
@@ -247,6 +247,59 @@ In Firefox's Web Console the [command line interpreter](https://developer.mozill
 - [REPL](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop)
 .....
 
+### Agent
+
+#### Definition
+An Agent is a thread that holds all state within a JavaScript VM instance.
```

Comment:
The definition of Agent in 8.7 makes it explicit that the agent is *not* a thread, but that multiple agents can share an executing thread.  The agent is an object containing all the state that an executing thread needs to evaluate the jobs that are on the agent's job queues.  (Also, the offered definition punts the problem of what an agent is by referring to a "JS VM instance", which isn't tremendously helpful.  By and large, an agent + an executing thread *is* a JS VM instance, so I don't think the latter term is needed.)

Reply:
@lars-t-hansen Thanks for the clarification. Does this work? 
```
An agent is an object contains the state (execution context) that an executing thread needs to evaluate the jobs that are in the agent's job queues. An agent and the executing thread are considered a JavaScript VM instance.
```

Reply:
An agent isn't an object.

Reply:
@young, you should remove the parenthetical "(execution context)" because an execution context is a technical term and execution contexts make up part of the state of the Agent, but not all of that state, see 8.7.  And as @jmdyck points out, technically the agent is not simply the object but the object plus the (often shared) execution thread.

How about "An agent is a pair: a container for a set of job queues and related execution state, and an execution thread.  The execution state is exclusive to the agent while the execution thread can be shared with other agents."  I strongly suggest you don't try to bring "VM instance" into this, because that's complicated. Lawyers will advice you to keep "JavaScript" out of it too ;-)

Reply:
> And as @jmdyck points out, technically the agent is not simply the object but the object plus the (often shared) execution thread.

It's not an object 'plus something' either. There is no object involved. (At least, not as the spec defines it.)

> How about "An agent is a pair: a container for a set of job queues and related execution state, and an execution thread. The execution state is exclusive to the agent while the execution thread can be shared with other agents."

The term "agent" is defined in the spec. Is there a need to define it here too?

In general, for terms defined in the spec, I grant that there might be reasons to add something here, but I'm not seeing those reasons in this case. Anyhow, there should be a link to the spec definition.

Reply:
> > And as @jmdyck points out, technically the agent is not simply the object but the object plus the (often shared) execution thread.
> 
> It's not an object 'plus something' either. There is no object involved. (At least, not as the spec defines it.)

I don't understand that.  An aspect of the agent clearly is its state.  Are you simply reacting to the word "object" above?  If not I wish you would explain yourself more thoroughly.

> > How about "An agent is a pair: a container for a set of job queues and related execution state, and an execution thread. The execution state is exclusive to the agent while the execution thread can be shared with other agents."
> 
> The term "agent" is defined in the spec. Is there a need to define it here too?
> 
> In general, for terms defined in the spec, I grant that there might be reasons to add something here, but I'm not seeing those reasons in this case. Anyhow, there should be a link to the spec definition.

So let the poster give his reason for including it here.  I don't really care either way whether the term is included in the glossary, but the initial summary provided here was clearly not capturing what an agent is.

Reply:
> Are you simply reacting to the word "object" above?

Yup.

Reply:
> The term "agent" is defined in the spec. Is there a need to define it here too?

@jmdyck The idea behind `terminology.md` is to define terms that might come up in plenary that may be opaque to newer delegates. I read the spec [definition](https://tc39.es/ecma262/#sec-agents), asked people in IRC, and received feedback here and even after all of that, I'm still not solid a on layman's definition of `agent`. @lars-t-hansen probably has the best one so far :).

Reply:
@lars-t-hansen How about this?
> An agent is a comprised of three things: a container for a set of job queues, related execution state, and an execution thread. The execution state is exclusive to the agent while the execution thread can be shared with other agents.

Reply:
Sure, sounds fine :-)